webapp/src/serial.tsx

```diff
@@ -224,30 +224,28 @@ export class Editor extends srceditor.Editor {
     }
 
     appendGraphEntry(source: string, variable: string, nvalue: number, receivedTime: number) {
-        this.loadSmoothieChartsAsync()
-            .then(() => {
-                //See if there is a "home chart" that this point belongs to -
-                //if not, create a new chart
-                let homeChart: Chart = undefined
-                for (let i = 0; i < this.charts.length; ++i) {
-                    let chart = this.charts[i]
-                    if (chart.shouldContain(source, variable)) {
-                        homeChart = chart
-                        break
-                    }
-                }
-                if (!homeChart) {
-                    homeChart = new Chart(source, variable, this.chartIdx, this.currentLineColors)
-                    this.chartIdx++;
-                    this.charts.push(homeChart)
-                    this.chartRoot.appendChild(homeChart.getElement());
-                    pxt.BrowserUtils.removeClass(this.chartRoot, "nochart");
-                    if (this.consoleRoot) {
-                        pxt.BrowserUtils.removeClass(this.consoleRoot, "nochart");
-                    }
-                }
-                homeChart.addPoint(variable, nvalue, receivedTime)
-            })
+        //See if there is a "home chart" that this point belongs to -
+        //if not, create a new chart
+        let homeChart: Chart = undefined
+        for (let i = 0; i < this.charts.length; ++i) {
+            let chart = this.charts[i]
+            if (chart.shouldContain(source, variable)) {
+                homeChart = chart
+                break
+            }
+        }
```

Comment:
```suggestion
        const homeChart = this.charts.find(chart => chart.shouldContain(source, variable));
```

This portions not new, but easy enough to cleanup; just personal preference though, up to you